custom_rules: Allow a line to be excluded multiple times.

amanagr opened the pull request:
For a regex, if an excluded line in present twice in a file, we
should be able to exclude it twice.

Conversation in https://chat.zulip.org/#narrow/stream/6-frontend/topic/realm.20creation.20form

andersk reviewed (changes requested):
We can’t just skip the `exclude_lines.remove`, because later there’s

```python
        if exclude_lines:
            print('Please remove exclusions for file %s: %s' % (fn, exclude_lines))
```

amanagr commented:
Okay! updated.

andersk commented:
I mean the `print()` wasn’t there for no reason, it’s there to stop us from accumulating dead exclusions. We should preserve this functionality, just correctly.

garg3133 commented:
I think instead of removing elements from the original set, we could add the matched elements to a new set and apply `difference_update` on both sets at last.

Or even better, we could make a copy of original set at the start and instead of removing elements from the original set, `discard` the matched elements from the second set. Then we can use the second set to print the dead exclusions.

amanagr committed:
custom_rules: Allow a line to be excluded multiple times.

For a regex, if an excluded line in present twice in a file, we
should be able to exclude it twice.

amanagr commented:
Yeah, makes sense. Thanks @andersk and @garg3133. Updated!

garg3133 commented on zulint/custom_rules.py:
```diff
@@ -109,14 +109,15 @@ def check_file_for_pattern(
         for (i, line, line_newline_stripped, line_fully_stripped) in line_tups:
             if line_fully_stripped in exclude_lines:
-                exclude_lines.remove(line_fully_stripped)
+                unmatched_exclude_lines.remove(line_fully_stripped)
```
I think `remove` should be replaced with `discard` as `remove` raises an error if the element is not present in the set which will be true while excluding the same line multiple times.

amanagr replied:
Cool! Thanks!
